ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/GroupByParamsFactory.java

```diff
@@ -44,48 +50,72 @@ private GroupByParamsFactory() {
 
   public static LogicalSchema buildSchema(
       final LogicalSchema sourceSchema,
-      final List<ExpressionMetadata> expressions
+      final List<ExpressionMetadata> groupBys,
+      final KsqlConfig ksqlConfig
   ) {
-    return expressions.size() == 1
-        ? singleExpressionSchema(sourceSchema, expressions.get(0).getExpressionType())
-        : multiExpressionSchema(sourceSchema);
+    return groupBys.size() == 1
+        ? singleExpressionSchema(sourceSchema, groupBys.get(0), ksqlConfig)
+        : multiExpressionSchema(sourceSchema, ksqlConfig);
   }
 
   public static GroupByParams build(
       final LogicalSchema sourceSchema,
-      final List<ExpressionMetadata> expressions,
-      final ProcessingLogger logger
+      final List<ExpressionMetadata> groupBys,
+      final ProcessingLogger logger,
+      final KsqlConfig ksqlConfig
   ) {
-    final Function<GenericRow, Struct> mapper = expressions.size() == 1
-        ? new SingleExpressionGrouper(expressions.get(0), logger)::apply
-        : new MultiExpressionGrouper(expressions, logger)::apply;
+    if (groupBys.isEmpty()) {
+      throw new IllegalArgumentException("No GROUP BY groupBys");
+    }
 
-    final LogicalSchema schema = buildSchema(sourceSchema, expressions);
+    final Function<GenericRow, Struct> mapper = groupBys.size() == 1
+        ? new SingleExpressionGrouper(groupBys.get(0), logger)::apply
+        : new MultiExpressionGrouper(groupBys, logger)::apply;
+
+    final LogicalSchema schema = buildSchema(sourceSchema, groupBys, ksqlConfig);
 
     return new GroupByParams(schema, mapper);
   }
 
   private static LogicalSchema multiExpressionSchema(
-      final LogicalSchema sourceSchema
+      final LogicalSchema sourceSchema,
+      final KsqlConfig ksqlConfig
   ) {
-    return buildSchemaWithKeyType(sourceSchema, SqlTypes.STRING);
+    return buildSchemaWithKeyType(sourceSchema, Optional.empty(), SqlTypes.STRING, ksqlConfig);
   }
 
   private static LogicalSchema singleExpressionSchema(
       final LogicalSchema sourceSchema,
-      final SqlType keyType
+      final ExpressionMetadata groupBy,
+      final KsqlConfig ksqlConfig
   ) {
-    return buildSchemaWithKeyType(sourceSchema, keyType);
+    final SqlType keyType = groupBy.getExpressionType();
+
+    final Optional<ColumnName> singleColumnName = Optional.of(groupBy.getExpression())
+        .filter(colRef -> ksqlConfig.getBoolean(KsqlConfig.KSQL_ANY_KEY_NAME_ENABLED))
+        .filter(gb -> gb instanceof ColumnReferenceExp)
+        .map(ColumnReferenceExp.class::cast)
+        .map(ColumnReferenceExp::getColumnName);
```

Comment:
👎 that is an abuse of `Optional`. Just making my opinion known, but as it's a style thing feel free to ignore me

Reply:
I agree - I'm being super lazy and this code is going very soon.  Next PR removes this.